apps/omg_api/lib/fresh_blocks.ex

```diff
@@ -56,9 +56,18 @@ defmodule OMG.API.FreshBlocks do
     {:reply, result, state}
   end
 
-  def handle_cast({:push, block}, state) do
+  def handle_cast({:push, %Block{transactions: [], number: block_number, hash: block_hash}}, state) do
```

Comment:
instead getting Block with an empty transactions, it should get :not_form or something else

Reply:
1. it's still misleading, because you return an empty block, but you don't persist it. If the name `form_block` is the problem, then perhaps `maybe_form_block`? to indicate the optionality. But I think `form_block` as a signal sent to an actor is still fine - `State` replies "no I won't" and that's it. The optionality is quite evident from the return type (list of Blocks, which might be empty) and a `@doc` can clarify any doubts further. Note, that with your approach you have `enqueue_block` and `push` which don't do what the verb says, so same problem. 
2. Good catch here, it might be the case that `wait_for_enqueue` is a leftover/redundant check :(. Let's check this

Reply:
due to problems with handling of empty blocks in the Watcher (`close_block`), let's consider reverting to your original idea with a pre-check in `BlockQueue`